### PR TITLE
Dialog with custom UI API 

### PR DESCRIPTION
**Pull request by llali**
The PR includes the API that extensions use to create a dialog with a custom html. 
the dialog is using a web view so there are changes in the webview code to support communicating  between the custom html and extension. those changes can also be used for when creating webview inside dashboard tabs to support custom UI in the tab.

### CODE REVIEW

File: src/sql/base/browser/ui/modal/webViewDialog.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (kburtram) [inline src/sql/base/browser/ui/modal/webViewDialog.ts]: Please use the "Source EULA" header in this repo.

File: src/sql/base/browser/ui/modal/webViewDialog.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import data = require('data');
+
+export class WebViewDialog extends Modal {
+
+	private _body: HTMLElement;
+	private _okButton: Button;
+	private _okLabel: string;
+	private _closeLabel: string;
+	private _webview: WebView;
+	public _html: string;
```
Comment (kburtram) [inline src/sql/base/browser/ui/modal/webViewDialog.ts]: the '_' is used to denote private field.  Please either make private or remove '_'.

File: src/sql/workbench/api/node/sqlExtHost.api.impl.ts [hunk 3]
```diff
@@ -247,7 +255,8 @@ export function createApiFactory(
 				MetadataType: sqlExtHostTypes.MetadataType,
 				TaskStatus: sqlExtHostTypes.TaskStatus,
 				TaskExecutionMode: sqlExtHostTypes.TaskExecutionMode,
-				ScriptOperation: sqlExtHostTypes.ScriptOperation
+				ScriptOperation: sqlExtHostTypes.ScriptOperation,
+				window: window
```
Comment (anthonydresser) [inline src/sql/workbench/api/node/sqlExtHost.api.impl.ts]: minor: you can just do `window` you don't need to do `window: window`

File: src/vs/workbench/parts/html/browser/webview-pre.js [hunk 4]
```diff
@@ -1,271 +1,282 @@
-/*---------------------------------------------------------------------------------------------
```
Comment (anthonydresser) [inline src/vs/workbench/parts/html/browser/webview-pre.js]: I'm assuming this converting tab spacing from spaces to tabs? can you confirm it's not the other way around.

File: src/vs/workbench/parts/html/browser/webview.ts [hunk 5]
```diff
@@ -1,466 +1,486 @@
...
-
-	public showNextFindTerm() {
-		this._webviewFindWidget.showNextFindTerm();
-	}
-
-	public showPreviousFindTerm() {
-		this._webviewFindWidget.showPreviousFindTerm();
-	}
-}
+/*---------------------------------------------------------------------------------------------
```
Comment (anthonydresser) [inline src/vs/workbench/parts/html/browser/webview.ts]: Same as the other file

**Pull request merged**